spatialdata/_core/_spatialdata_ops.py

```diff
@@ -281,3 +288,165 @@ def _describe_paths(paths: list[list[Union[int, str]]]) -> str:
             transformations.append(g[path[i]][path[i + 1]]["transformation"])
         sequence = Sequence(transformations)
         return sequence
+
+
+##
+def concatenate_tables(tables: list[AnnData]) -> Optional[AnnData]:
+    """
+    Concatenate a list of tables using AnnData.concatenate() and preserving the validity of region, region_key and instance_key
+
+    Parameters
+    ----------
+    tables
+        A list of tables to concatenate
+
+    Returns
+    -------
+    A table with all the tables concatenated. If the list of tables is empty, None is returned.
+
+    Notes
+    -----
+    Not all tables can be merged, they need to have compatible region, region_key and instance_key values. This function
+    checks this and merges if possible.
+
+    """
+    if len(tables) == 0:
+        return None
+    if len(tables) == 1:
+        return tables[0]
+
+    MERGED_TABLES_REGION_KEY = "annotated_element_merged"
```

Comment:
I think this should be also available in the signature for the user to choose

Reply:
Since `concatenate_table()` will be private we would have to expose this in `concatenate()`, but it's a tricky parameter, I am not sure we want the user to have to think about this when calling the method: sometimes a new column is added, other times the `REGION_KEY` value can be kept the one that was already present. In the comment below where I explain the code with the for loop I have explained that some edge cases have been left out. I would wait until we address those cases, and in that moment I would expose `MERGED_TABLES_REGION_KEY`, since its logic would change. wdyt?